server/src/test/java/io/crate/expression/reference/doc/lucene/SourceParserTest.java

```diff
@@ -350,16 +350,45 @@ public void test_drop_sub_column_with_children() {
             .setInnerType("b", DataTypes.INTEGER)
             .build();
 
-
+        // Register parent column in order to collect ony this one, ignoring any other column
         sourceParser.register(new ColumnIdent("_doc", List.of("o")), oType);
         Map<String, Object> result = sourceParser.parse(new BytesArray(
             """
-                {"o": {"a" : 1, "b": 2, "oo": {"a": 11, "b": 22, "c": 3}}}
+                {"o": {"a" : 1, "b": 2, "oo": {"a": 11, "b": 22, "c": 3}}, "x": 4}
+            """));
+
+        assertThat(Maps.getByPath(result, "o.a")).isEqualTo(1);
+        assertThat(Maps.getByPath(result, "o.b")).isEqualTo(2);
+        assertThat(Maps.getByPath(result, "o.oo")).isNull();
+        assertThat(Maps.getByPath(result, "x")).isNull();
+    }
+
+    @Test
+    public void test_drop_sub_column_with_children_collect_all() {
+        SourceParser sourceParser = new SourceParser(
+                Set.of(createReference(new ColumnIdent("o", List.of("oo")), DataTypes.INTEGER)),
```

Comment:
Just for clarity: this should be ObjectType, no?
so maybe first create ooType, and use it here instead of `DataTypes.INTEGER`.

Same is for the previous test.

Reply:
Good catch! will fix